Library/Homebrew/brew.sh

```diff
@@ -76,7 +76,11 @@ export GEM_OLD_PATH="$GEM_PATH"
 # Users may have these set, pointing the system Ruby
 # at non-system gem paths
 unset GEM_HOME
-unset GEM_PATH
```

Comment:
I'd just remove both these `unset` and let your script handle them. You can check through all the `unset` in this file and remove them all, I think.